src/Employer/Employer.Web/Orchestrators/Part1/DatesOrchestrator.cs

```diff
@@ -98,6 +103,20 @@ public async Task<OrchestratorResponse> PostDatesEditModelAsync(DatesEditModel m
             );
         }
 
+        private async Task SetTrainingProgrammeAsync(Vacancy vacancy, DatesViewModel vm)
+        {
+            if (string.IsNullOrEmpty(vacancy.ProgrammeId))
+                return;
+
+            var programme = await _apprenticeshipProgrammeProvider.GetApprenticeshipProgrammeAsync(vacancy.ProgrammeId);
+
+            if (programme?.EffectiveTo.HasValue == true)
+            {
+                vm.TrainingDescription = $"{programme.Title}, Level:{(int)programme.Level}";
+                vm.TrainingEffectiveToDate = programme.EffectiveTo.Value.AsGdsDate();
```

Comment:
What if EffectiveTo is null? this will go bang I think. `AsGdsDate` does not check

Reply:
3 lines above there is a null check.